apps/mobile/src/components/sheets/warning-sheet.layout.tsx

```diff
@@ -50,7 +53,9 @@ export function WarningSheet({
                 >
                   <ErrorTriangleIcon
                     color={
-                      variant === 'critical' ? 'red.action-primary-default' : 'ink.text-primary'
+                      theme.colors[
```

Comment:
Maybe we could move this into a const a bit higher up to make it a bit easier to read? 

Having a ternary inside an array lookup is a bit tricky.

This could be nicer:
```
const iconColor = variant === 'critical' ? theme.colors['red.action-primary-default'] : theme.colors['ink.text-primary'];
...
  <ErrorTriangleIcon
                    color={iconColor}
```

This icon could probably even be in a separate wrapped component that loads the `useTheme` hook there. 

Looking at this code and I see the `Box` has this color settings:
```
<Box
                  bg={
                    variant === 'critical' ? 'red.background-primary' : 'ink.background-secondary'
                  }
```

1. Do we only need the `theme[...]` syntax for the icon?
2. Maybe we can just have this `Box` as a component accepting `variant`